lib/Doctrine/DBAL/Driver/SQLite3/SQLite3Statement.php

```diff
@@ -0,0 +1,394 @@
+     */
+    private $lastErrorCode = 0;
+
+    /**
+     * The error message for the last execution of the statement.
+     *
+     * @var string
+     */
+    private $lastErrorMessage = 'not an error';
+
+    /**
+     * Class constructor.
+     *
+     * @param \SQLite3     $sqlite3 The SQLite3 connection object.
+     * @param \SQLite3Stmt $stmt    The SQLite3 statement object.
+     */
+    public function __construct(\SQLite3 $sqlite3, \SQLite3Stmt $stmt)
+    {
+        $this->sqlite3 = $sqlite3;
+        $this->stmt    = $stmt;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function bindValue($param, $value, $type = null)
+    {
+        try {
+            if ($type === null) {
+                return $this->stmt->bindValue($param, $value);
+            }
+
+            return $this->stmt->bindValue($param, $value, $this->convertType($type));
+        } catch (\Exception $e) {
+            throw $this->wrapDriverException($e);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function bindParam($column, & $variable, $type = null, $length = null)
+    {
+        try {
+            if ($type === null) {
+                return $this->stmt->bindParam($column, $variable);
+            }
+
+            return $this->stmt->bindParam($column, $variable, $this->convertType($type));
+        } catch (\Exception $e) {
+            throw $this->wrapDriverException($e);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function errorCode()
+    {
+        return $this->lastErrorCode;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function errorInfo()
+    {
+        return [
+            null,
+            $this->lastErrorCode,
+            $this->lastErrorMessage
+        ];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function execute($params = null)
+    {
+        try {
+            if ($params) {
+                foreach ($params as $key => $value) {
+                    $this->bindValue(is_int($key) ? $key + 1 : $key, $value);
+                }
+            }
+
+            $result = $this->stmt->execute();
+
+            $this->lastErrorCode = $this->sqlite3->lastErrorCode();
+            $this->lastErrorMessage = $this->sqlite3->lastErrorMsg();
+
+            $this->result = $result;
+            $this->rowCount = $this->sqlite3->changes();
+        } catch (\Exception $e) {
+            throw $this->wrapDriverException($e);
+        }
+
+        $this->throwExceptionOnError();
+
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function rowCount()
+    {
+        return $this->rowCount;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function closeCursor()
+    {
+        return $this->result->finalize();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function columnCount()
+    {
+        if ($this->result) {
+            try {
+                return $this->result->numColumns();
```

Comment:
This method can throw an exception if the SQLite3Stmt initialization check fails. 

See: https://github.com/php/php-src/blob/master/ext/sqlite3/sqlite3.c#L1654